src/nodes/chunk_dispatch/chunk_dispatch.c

```diff
@@ -164,16 +164,25 @@ ts_chunk_dispatch_decompress_batches_for_insert(ChunkDispatch *dispatch, ChunkIn
 {
 	if (cis->chunk_compressed)
 	{
+		OnConflictAction onconflict_action = chunk_dispatch_get_on_conflict_action(dispatch);
+		Oid amoid = get_table_am_oid("hyperstore", false);
+
+		if (cis->rel->rd_rel->relam == amoid && onconflict_action != ONCONFLICT_UPDATE)
```

Comment:
@antekresic Feel free to suggest a concrete change. Otherwise I will leave this as is, with possibility to refactor later.

Reply:
We've submitted commits to fix flaky tests and other things as separate PRs. I believe these were the "unrelated" commits.

Reply:
Its hard to suggest an actual change here but the idea is to push the check into `chunk_dispatch_exec` and `copyfrom` functions since AM doesn't change with every slot that's being inserted hence you only need to check it once and avoid calling `ts_chunk_dispatch_get_on_conflict_action` altogether.

Reply:
Changed this to cache whether TAM is used in chunk_insert_state.